src/algo/mod.rs

```diff
@@ -355,7 +353,8 @@ impl<N> TarjanScc<N> {
     /// Creates a new `TarjanScc`
     pub fn new() -> Self {
         TarjanScc {
-            index: 0,
+            index: std::usize::MAX, // Invariant: index > componentcount at all times.
```

Comment:
Why does index count downwards instead of upwards from 1, as in the paper's pseudocode?

Reply:
Because that lets you have componentcount counting upwards, which makes it possible to do `self.nodes[g.to_index(v)] == Some(k)` when v belongs to the kth component. On the other hand, if we're going for the `Option<NonZeroUsize>` route and need a subtraction anyway since zero is ruled out, we can just flip them back to the original setup and have  `self.nodes[g.to_index(v)] == Some(usize::MAX - k)` instead.

Reply:
Okay, Rustc optimizes usize::MAX - k properly to the ones complement of k, so that also basically costs nothing.
https://godbolt.org/z/GM8z7neEK

Reply:
Great, that works then.